pkg/callback/scheduler_callback.go

```diff
@@ -149,6 +149,16 @@ func (callback *AsyncRMCallback) RecvUpdateResponse(response *si.UpdateResponse)
 			if err != nil {
 				log.Logger().Error("failed to delete application", zap.Error(err))
 			}
+		} else if updated.State == events.States().Application.Resuming {
+			app := callback.context.GetApplication(updated.ApplicationID)
+			if app != nil && app.GetApplicationState() == events.States().Application.Reserving {
+				ev := cache.NewRunApplicationEvent(updated.ApplicationID)
+				dispatcher.Dispatch(ev)
+
+				// handle status update
+				dispatcher.Dispatch(cache.NewApplicationStatusChangeEvent(updated.ApplicationID, events.AppStateChange, updated.State))
+			}
```

Comment:
The core side will move the application into the Resuming state before we have all the placeholders removed. See the changes in the PR: https://github.com/apache/incubator-yunikorn-core/pull/275/files

I think when we need to move this part to when we remove the allocations and asks, to make sure that first we delete all the placeholders.